File: 71. Simplify Path/main.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = list()
        words = path.split(sep="/")

        for word in words:
            if word == "..":
                if len(stack) != 0:
                    stack.pop()
            else:
                if word != "" and word != ".":
                    stack.append(word)

        result = ""
        for word in stack:
            result += "/"
            result += word

        if len(stack) == 0:
            result += "/"

        return result


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = list()
        words = path.split(sep="/")

        for word in words:
            if word == "..":
                if stack:
                    stack.pop()
            else:
                if word and word != ".":
                    stack.append(word)

        result = ""
        for word in stack:
            result += "/"
            result += word

        if len(stack) == 0:
            result += "/"

        return result


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = list()
        words = path.split(sep="/")

        for word in words:
            if word == "..":
                if stack:
                    stack.pop()
            else:
                if word and word != ".":
                    stack.append(word)

        return "/" + "/".join(stack)

File: 71. Simplify Path/main.rs
struct Solution;

impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut stack = Vec::new();

        for dir in path.split("/") {
            if dir == ".." {
                stack.pop();
            } else {
                if !dir.is_empty() && dir != "." {
                    stack.push(dir);
                }
            }
        }

        format!("/{}", stack.join("/"))
    }
}

impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut stack = Vec::new();

        for dir in path.split("/") {
            match dir {
                "" => {
                    continue;
                }
                "." => {
                    continue;
                }
                ".." => {
                    stack.pop();
                }
                s => {
                    stack.push(s);
                }
            }
        }

        format!("/{}", stack.join("/"))
    }
}
